src/main/java/database/DataInstancesBuilder.java

```diff
@@ -0,0 +1,119 @@
+package database;
+
+import utils.RandomString;
+import weka.core.Attribute;
+import weka.core.FastVector;
+import weka.core.Instance;
+import weka.core.Instances;
+
+import javax.xml.crypto.Data;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.*;
+
+/**
+ * Created by Timotej on 31-Oct-15.
+ *
+ * This class builds up weka dataset (Instances)
+ */
+public class DataInstancesBuilder {
+
+    private int batchID = -1;
+    private int subjectID = -1;
+    private Map<Integer, String> features = new HashMap<Integer, String>();
+    private RandomString rndStr = new RandomString((int) Math.ceil(Math.pow(
+            features.size(),
+            1 / RandomString.alphabetSize())) + 1);
+
+    /**
+    REQUIRED tells which batch to use
+     */
+    public DataInstancesBuilder forBatch(int batchID) {
+        this.batchID = batchID;
+        return this;
+    }
+
+    /**
+    OPTIONAL, if used it just gets data for that subject
+     */
+    public DataInstancesBuilder forSubject(int subjectID) {
+        this.subjectID = subjectID;
+        return this;
+    }
+
+    /**
+     REQUIRED tells which features to put into the dataset
+     Call multiple times to add more features
+     */
+    public DataInstancesBuilder withFeature(int featureID) {
+        features.put(featureID, rndStr.nextString());
+        return this;
+    }
+
+
+
+    public Instances build(Statement stmt) {
+        int numFeatures = features.size();
+        assert numFeatures > 0 : "need atleast one feature";
+        Instances dataset;
+
+        try {
+            String fields = "";
+            String joins = "";
+            String ands = "";
+            String currentAlias;
+            boolean first = true;
+
+            FastVector attributes = new FastVector(numFeatures + 1);
+            final Attribute subjectIdAttribute = new Attribute("subjectID");
+            attributes.addElement(subjectIdAttribute);
+
+            Map<Integer, Attribute> attributeMap = new HashMap<Integer, Attribute>();
+
+            for (Map.Entry<Integer, String> feature : features.entrySet()) {
+                currentAlias = feature.getValue();
+                Attribute a = new Attribute(feature.getKey().toString());
+                attributes.addElement(a);
+                attributeMap.put(feature.getKey(), a);
+
+                if (first) {
+                    first = false;
+                    fields += currentAlias + ".subjectID";
+                    joins += " data as " + currentAlias;
+                } else {
+                    joins += " JOIN data as " + currentAlias + " USING (subjectID, batchID)";
+                }
+
+                fields += ", " + currentAlias + ".featureValue AS " + currentAlias + "fv";
+
+                ands += " AND " + currentAlias + ".featureID = " +  feature.getKey();
+
+            }
+            ResultSet rs = stmt.executeQuery("SELECT " + fields +
```

Comment:
What are you talking about? There is no possiblity of SQL injection. All the stuff used in the concatenation of strings is built from private fields of the class. Also it's never used in the public side and there are tests for it.